### PR TITLE
New-DbaDatabaseSnapshot tests fixes

### PR DESCRIPTION
**Pull request by niphlod**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [ ] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [x] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 
### Purpose
Notes are hidden if no exception occurs. Until latest SMO, it was always the case. 
It seems that with the latest SMO no weird workarounds are needed (someone in MS is listening in dbatools channel, apparently :-P ). The check for default props has been adjusted to look for the common denominator of default properties.

tl;dr: my faith in computer systems is restored ^_^

### CODE REVIEW

**Comment by potatoqualitee** [comment]
gorgeocity of 10
